### DISCUSSION TITLE
[Review] Reviews not working (Is Actions blocking Reviews?)

### DISCUSSION THREAD

**Discussion by ghost** [community/community#127146] [Actions] [answered]
### Select Topic Area

Question

### Body

What can I do to get help with GitHub Actions? I don't have enough time to search for similar discussions. Should I just stop using GitHub Actions? Some of them say they are supposed to just work by pressing the install new actions button and not changing the code. Rulesets are also an issue like this. How is it that rulesets don't work right when all you do is select a checkbox and hit save? There should be no need for me to search for similar discussion. There's nothing I can do to fix this, except stop using the particular GitHub Actions and Rulesets that are malfunctioning.

**Comment by Toandz1125** [comment]
Response #1

**Reply by ghost** [reply]
> Response #1

Thank you for your response. Is there any more you can provide? I'm not using any of the services you mentioned.

**Comment by ghost** [comment]
One of the replies got moderated apparently by someone who has not participated in this discussion. Because this is a discussion, and considering that the moderated responder has less than 100 commits, I'm going to offer a little more in the scope of feedback. Some may choose to go to use code spaces and some other more advanced GitHub tools if they are not satisfied with the limitations of GitHub actions. I have a different philosophy that I typically apply to products and tools that I use. 

I want to see where I can find a fundamental working ground with some of the main features of GitHub. Actions is in the top header bar and reviews are part of the Pull Request functionality. When I find a good rudimental working usage of these to features, then I will begin to explore the possibility of whether or not to advance to the next needed feature.

I suppose the next reply I would write, if I get time, would be one that uses content from similar discussions as reference.

**Reply by ghost** [reply]
> One of the replies got moderated apparently by someone who has not participated in this discussion. Because this is a discussion...

I'm not sure if the person that replied moderated themselves out of this discussion or not. I have been learning more about feedback loops and I do want to try to accept this other bias.

GitHub Action connection points like GitHub Codespaces is definitely interesting and there may be some people that are interested in including these other GitHub features because they can connect directly to GitHub Actions and really deserve attention in this particular topic.

**Comment by ghost** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> What can I do to get help with GitHub Actions? I don't have enough time to search for similar discussions. Should I just stop using GitHub Actions? Some of them say they are supposed to just work by pressing the install new actions button and not changing the code. Rulesets are also an issue like this. How is it that rulesets don't work right when all you do is select a checkbox and hit save? There should be no need for me to search for similar discussion. There's nothing I can do to fix this, except stop using the particular GitHub Actions and Rulesets that are malfunctioning.

Reviews is simply not working. There may be some combination of Actions and rulesets reviews that works, but it's just not working as expected. I am in favor of looking for a different method for managing pull request details.

**Comment by ghost** [comment]
_At least 1 approving review is required by reviewers with write access._

This message showed up for the first time. My process of reducing the use of actions and rulesets down appears to have summoned this more helpful message. Now all it appears I have to do is give my reviewer write access. how do i give write access to reviewers?

**Reply by ghost** [reply]
> _At least 1 approving review is required by reviewers with write access._
> 
> This message showed up for the first time. My process of reducing the use of actions and rulesets down appears to have summoned this more helpful message. Now all it appears I have to do is give my reviewer write access. how do i give write access to reviewers?

It works now. You have to go to Collaborators in the settings page on the repository and add the reviser as the collaborator. This is worked after simplifying the actions and rulesets down until receiving a clear and uncluttered message on why the pull request is not able to be merged.

**Reply by ghost** [reply]
It's important to add that this strange reviews and actions behavior still doesn't make any sense logically speaking. It actually wasn't that the reviewer was not a collaborator that caused the reviews to not work. That wasn't the issue. It appears to have fixed the issue, though. Reviews were working okay for a while, before I ever added any collaborators. I can't see much logic in this actions and reviews pull request behavior. I am still hesitant to depend on these features.

**Reply by ghost** [reply, marked as answer]
I've deactivate reviews and the collaboration features. this may be a better solution than the previous one marked. I'm considering deactivating rulesets and actions altogether for a better experience on GitHub. I don't have time to research GitHub documentation, so some other GitHub people are free to guide me with this one if they want to. The lack of consistency is too expensive for me right now.

**Comment by ghost** [comment]
Reviews feature should not be requiring to authorize reviewers to be given write permissions. That doesn't make sense. It sounds like a security vulnerability on GitHub. I'm moving more of my repositories out of GitHub. I looked through some documentation and this write permissions philosophy appears to be stronger than allowing the repository to restrict reviewers from making changes.

**Reply by ghost** [reply]
> Reviews feature should not be requiring to authorize reviewers to be given write permissions. That doesn't make sense. It sounds like a security vulnerability on GitHub. I'm moving more of my repositories out of GitHub. I looked through some documentation and this write permissions philosophy appears to be stronger than allowing the repository to restrict reviewers from making changes.

Is GitHub operating from open source code that has not been fully reviewed for some features?

**Reply by ghost** [reply]
> > Reviews feature should not be requiring to authorize reviewers to be given write permissions. That doesn't make sense. It sounds like a security vulnerability on GitHub. I'm moving more of my repositories out of GitHub. I looked through some documentation and this write permissions philosophy appears to be stronger than allowing the repository to restrict reviewers from making changes.
> 
> Is GitHub operating from open source code that has not been fully reviewed for some features?

I don't know, but here's (Why are our actions being stopped? #66522) what looks like an ubuntu update that was run at the same moment I encountered the issues that look like they may coming from Actions. It appears some runner stopped, and likewise, I am still not changing my review that there may be some philosophical strangeness concerning the security of GitHub costumers proprietary code.
